src/groupByKey.ts

```diff
@@ -2,6 +2,14 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemValue = String(item[key]);
+
+    groupedItems[itemValue] = items.filter((obj) => obj[key] === item[key]);
```

Comment:
This line is currently re-filtering the `items` array for every item, which is inefficient as it results in O(n^2) complexity. Instead, you should check if `groupedItems[itemValue]` already exists; if it doesn't, initialize it with an empty array, and then simply push the `item` onto that array. This will reduce the complexity to O(n).